pkgs/os-specific/linux/switcheroo-control/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, ninja, meson, fetchFromGitLab, systemd, libgudev, pkgconfig, glib, python3Packages, gobject-introspection }:
+python3Packages.buildPythonApplication rec {
+  pname = "switcheroo-control";
+  version = "2.3";
+  src = fetchFromGitLab {
+    hash = "sha256-1Pze2TJ9mggfcpiLFwJ7/9WhsdJx4G3GoA7+Z47shuc=";
+    domain = "gitlab.freedesktop.org";
+    owner = "hadess";
+    repo = pname;
+    rev = version;
+  };
+  nativeBuildInputs = [
+    ninja meson pkgconfig glib
+  ];
+  buildInputs = [
+    systemd libgudev
+  ];
+  propagatedBuildInputs = [ python3Packages.pygobject3 ];
+  mesonFlags = [ "-Dsystemdsystemunitdir=${placeholder "out"}/etc/systemd/system" "-Dhwdbdir=${placeholder "out"}/etc/udev/hwdb.d"];
+  format = "other";
+  meta = with lib; {
+    description = "D-Bus service to check the availability of dual-GPU";
+    homepage = "https://gitlab.freedesktop.org/hadess/switcheroo-control/";
+    changelog = "https://gitlab.freedesktop.org/hadess/switcheroo-control/-/blob/${version}/NEWS";
+    license = licenses.gpl3;
```

Comment:
`gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.